csrc/id_model/indexing.cpp

```diff
@@ -884,8 +884,8 @@ std::vector<Val*> TensorIndexer::getIndexFor(
     auto it = info.index_map.find(g);
     NVF_ERROR(
         it != info.index_map.end(), "Index not found for ", g->toString());
-    result.push_back(
-        ir_utils::replaceValRecursively(it->second, replacement_map));
+    result.push_back(simplifyExpr(
```

Comment:
Is this `simplifyExpr` strictly required?

Reply:
I looked at the cuda kernels from the modified tests and there wasn't a difference when I removed `simplifyExpr`. The expressions are likely simplified by scalar hoist pass.

It isn't necessary for the TMA circular buffering tests.

Reply:
Thanks. That was what I thought - since these vals are simplified later, it should not be necessary to run `simplifyExpr` here, which can be costly unlike `SimplifyingIrBuilder`.

Reply:
Let's remove the `simplifyExpr`.